src/main/scala/org/constellation/infrastructure/endpoints/middlewares/PeerAuthMiddleware.scala

```diff
@@ -52,22 +54,58 @@ object PeerAuthMiddleware {
     }
 
   def responseSignerMiddleware[F[_]: Sync](
-    privateKey: PrivateKey
+    privateKey: PrivateKey,
+    sessionTokenService: SessionTokenService[F]
   )(http: HttpRoutes[F])(implicit C: ContextShift[F]): HttpRoutes[F] =
     Kleisli { req: Request[F] =>
-      http(req).flatMap { res =>
-        new Http4sResponseSigner[F](getGenerator(privateKey))
-          .sign(res)
+      val logger = Slf4jLogger.getLogger[F]
+
+      for {
+        res <- http(req)
+        oHeaderToken <- sessionTokenService.getOwnToken
+          .map(_.map(t => Header(XSessionToken.value, t.value)))
           .attemptT
           .toOption
+        newHeaders = oHeaderToken.fold(res.headers)(h => res.headers.put(h))
+        resWithHeader = res.copy(headers = newHeaders)
```

Comment:
actually you can chain it into one computation,
```
sessionTokenService.getOwnToken()
  .map {
    _.map(Header(XSessionToken.value, _.value)))
      .fold(res.headers)(res.headers.put)
      .map(h => res.copy(headers = h))
  }
```
or actually sth like that. just writing it by hand without type support so can be wrong

Reply:
Will keep it as it is after refactoring it's not very readable in my opinion.